events/api.py

```diff
@@ -3119,8 +3152,13 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
     # Filter by free offer
     val = params.get("is_free", None)
     if val and val.lower() in ["true", "false"]:
+        # Include events that have at least one free offer
         if val.lower() == "true":
-            queryset = queryset.filter(offers__is_free=True)
+            queryset = queryset.filter(
+                Exists(Offer.objects.filter(event=OuterRef("pk"), is_free=True))
+            )
+            queryset = queryset.exclude(~Q(offers__is_free=True))
```

Comment:
Isn't this line the same as the one above, just with the NOT NOT? :thinking:

Reply:
whoops!